public/app/plugins/datasource/loki/querybuilder/LokiQueryModeller.ts

```diff
@@ -18,12 +23,65 @@ export class LokiQueryModeller extends LokiAndPromQueryModellerBase {
     ]);
   }
 
-  renderLabels(labels: QueryBuilderLabelFilter[]) {
+  renderOperations(queryString: string, operations: QueryBuilderOperation[]): string {
+    for (const operation of operations) {
+      const def = this.operationsRegistry.getIfExists(operation.id);
+      if (!def) {
+        throw new Error(`Could not find operation ${operation.id} in the registry`);
```

Comment:
This is great. Maybe, maybe, I would recommend for good measure to `console.error` and not throw, because in the rare event that this happens, it might render the app unusable until the user refreshes, so maybe:

```
console.error('...');
continue; or return;
```

Reply:
Great point, fixed in https://github.com/grafana/grafana/pull/81140/commits/ce379b98ab30ec2d0d0b6f4484fc5da3cf91af08!